File: rust/src/models/batch_job_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{
    batch_job_id::BatchJobId, batch_job_status::BatchJobStatus, batch_job_type::BatchJobType,
};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct BatchJobResponse {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub completed_at: Option<String>,

    pub created_at: String,

    pub created_by: String,

    pub failed_items: i32,

    pub id: BatchJobId,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub input_file_name: Option<String>,

    pub job_type: BatchJobType,

    pub processed_items: i32,

    pub status: BatchJobStatus,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub total_items: Option<i32>,
}

impl BatchJobResponse {
    pub fn new(
        created_at: String,
        created_by: String,
        failed_items: i32,
        id: BatchJobId,
        job_type: BatchJobType,
        processed_items: i32,
        status: BatchJobStatus,
    ) -> Self {
        Self {
            completed_at: None,
            created_at,
            created_by,
            failed_items,
            id,
            input_file_name: None,
            job_type,
            processed_items,
            status,
            total_items: None,
        }
    }
}

File: typescript/src/models/batchJobResponse.ts
// this file is @generated
import { type BatchJobId, BatchJobIdSerializer } from "./batchJobId";
import { type BatchJobStatus, BatchJobStatusSerializer } from "./batchJobStatus";
import { type BatchJobType, BatchJobTypeSerializer } from "./batchJobType";

export interface BatchJobResponse {
  completedAt?: Date | null;

  createdAt: Date;

  createdBy: string;

  failedItems: number;

  id: BatchJobId;

  inputFileName?: string | null;

  jobType: BatchJobType;

  processedItems: number;

  status: BatchJobStatus;

  totalItems?: number | null;
}

export const BatchJobResponseSerializer = {
  _fromJsonObject(object: any): BatchJobResponse {
    return {
      completedAt:
        object["completed_at"] != null ? new Date(object["completed_at"]) : undefined,
      createdAt: new Date(object["created_at"]),
      createdBy: object["created_by"],
      failedItems: object["failed_items"],
      id: BatchJobIdSerializer._fromJsonObject(object["id"]),
      inputFileName: object["input_file_name"],
      jobType: BatchJobTypeSerializer._fromJsonObject(object["job_type"]),
      processedItems: object["processed_items"],
      status: BatchJobStatusSerializer._fromJsonObject(object["status"]),
      totalItems: object["total_items"],
    };
  },

  _toJsonObject(self: BatchJobResponse): any {
    return {
      completed_at: self.completedAt,
      created_at: self.createdAt,
      created_by: self.createdBy,
      failed_items: self.failedItems,
      id: BatchJobIdSerializer._toJsonObject(self.id),
      input_file_name: self.inputFileName,
      job_type: BatchJobTypeSerializer._toJsonObject(self.jobType),
      processed_items: self.processedItems,
      status: BatchJobStatusSerializer._toJsonObject(self.status),
      total_items: self.totalItems,
    };
  },
};
